patches/server/0968-New-event-verbose-logging-command.patch

```diff
@@ -0,0 +1,264 @@
+@@ -45,6 +45,7 @@ public final class PaperCommand extends Command {
+         commands.put(Set.of("dumpitem"), new DumpItemCommand());
+         commands.put(Set.of("mobcaps", "playermobcaps"), new MobcapsCommand());
+         commands.put(Set.of("dumplisteners"), new DumpListenersCommand());
++        commands.put(Set.of("dumpevents"), EventVerboseCommand.get());
+ 
+         return commands.entrySet().stream()
+             .flatMap(entry -> entry.getKey().stream().map(s -> Map.entry(s, entry.getValue())))
+diff --git a/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java b/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1b6efa0454c4504c92048ddfaa20fc727105a8bd
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/command/subcommands/EventVerboseCommand.java
+@@ -0,0 +1,148 @@
++package io.papermc.paper.command.subcommands;
++
++import com.destroystokyo.paper.event.server.ServerTickEndEvent;
++import com.destroystokyo.paper.event.server.ServerTickStartEvent;
++import com.google.gson.JsonArray;
++import com.google.gson.JsonObject;
++import com.google.gson.internal.Streams;
++import com.google.gson.stream.JsonWriter;
++import io.papermc.paper.command.PaperSubcommand;
++import io.papermc.paper.util.misc.EventWrapper;
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.command.CommandSender;
++import org.bukkit.event.Event;
++import org.bukkit.inventory.ItemStack;
++import org.checkerframework.checker.nullness.qual.NonNull;
++import org.checkerframework.checker.nullness.qual.Nullable;
++import org.checkerframework.framework.qual.DefaultQualifier;
++
++import java.io.IOException;
++import java.io.PrintStream;
++import java.io.StringWriter;
++import java.lang.reflect.Field;
++import java.lang.reflect.Modifier;
++import java.nio.charset.StandardCharsets;
++import java.nio.file.Files;
++import java.nio.file.Path;
++import java.time.LocalDateTime;
++import java.time.format.DateTimeFormatter;
++import java.util.ArrayList;
++import java.util.List;
++import java.util.Map;
++import java.util.Collection;
++
++import static net.kyori.adventure.text.Component.text;
++import static net.kyori.adventure.text.format.NamedTextColor.GREEN;
++import static net.kyori.adventure.text.format.NamedTextColor.RED;
++
++@DefaultQualifier(NonNull.class)
++public class EventVerboseCommand implements PaperSubcommand {
++
++    @Nullable
++    private static EventVerboseCommand instance; //instance is declared registering in PaperCommand
++    private final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH.mm.ss");
++
++    private final List<Wrapper> WRAPPERS = new ArrayList<>();
++    private final Path PARENT = Path.of("event-debug");
++    //ignore unnecessary events
++    private final List<Class<? extends Event>> EXCLUDE = List.of(ServerTickStartEvent.class,
```

Comment:
Use a set (although it may not matter currently due to its hyper-optimized implemention... but future planning)